CONTRIBUTING.md

```diff
@@ -0,0 +1,131 @@
+your code has been properly formatted (can be done locally with `make format`),
+builds (`make` and `make check`), and passes the unit tests (`make test`).
+The CI will run in the branches of your fork, so you can verify it passes
+there before opening a Pull Request.
+
+## Testing
+
+Describe what kind of testing your contribution underwent. If the testing was
+manual, please describe the commands or external clients that were used. If the
+tests were automated, include at least a cursory description/name of the tests,
+when they were regressed, if possible.
+
+Please note that any contribution to the code of the project will likely
+require at least some testing to be done. Depending on how central the
+component touched in your PR is to the system, the more things could only be
+detected in real-world usecases through integration testing.
+
+If a commit has been determined to break integration testing at a later stage,
+please note that the first and safest measure will almost always be reverting
+the faulty commit. Making sure critical tests are passing remains a priority
+over waiting for some commit to be reworked or refactored (which can be worked
+on after a revert has been done). Though we are striving to make more tests
+public (with failure then being visible to all), as long as some critical tests
+remain private, this will also apply to such tests (with maintainers flagging
+the breakage preferrably describing at least the gist of the test).
+
+If you are still waiting on some testing to be done, please mark the PR as a
+"draft" and make the reasoning clear.
+
+If your contribution doesn't intend to have any functional changes, please make
+that clear as well.
+
+## Feature work
+
+If your contribution adds some new feature or reworks some major aspect of the
+system (as opposed to one-off fixes), it can be benefitial to first describe
+the plan of your work in a design proposal. Architectural issues are better
+spotted early on, and taking a big-picture view can often lead to new insights.
+
+An example of a design proposal is here:
+
+https://github.com/xapi-project/xen-api/pull/6387
+
+If submitting a design first is not possible, include documentation alongside
+with your PR describing the work, like it was done in the last three commits
+here:
+
+https://github.com/xapi-project/xen-api/pull/6457
+
+Note that the design will often serve as documentation as well - so take care
+updating it after the implementation is done to better reflect reality.
+
+## Review process and merge
+
+It can often be useful to address review suggestions with a "fixup" commit
+(created manually or with the help of `git commit --fixup=HASH`). This way it
+is clear what the original code was and what your fix touches. Once the
+fixup commit has been reviewed and the PR approved, please squash the fixup
+commits with `git rebase --autosquash` before merging. Otherwise the commits in
```

Comment:
I don't think Jujutsu supports fixup commits yet, but I think an equivalent workflow for those that use Jujutsu for everything (like me) would be to create a new commit with `jj new`, and then use `jj absorb` at the end to split up that commit and squash it into other commits in the PR as far up the commit history as possible.

Although there is no clear way to indicate on github whether the fixup commit has been reviewed yet or not, and when it is safe to squash it back in.